### ISSUE TITLE
Bogus error when using `..` in struct expr with all fields present, in a const ("error[E0016]: blocks in constants are limited to items and tail expressions")

### ISSUE THREAD

**Issue by jorendorff** [rust-lang/rust#39161]
```rust
struct X { a: i32, b: i32 }

#[allow(unused_variables)]
fn main() {
    const DX: X = X { a: 0, b: 0 };
    const X1: X = X { a: 1, ..DX };  // ok
    let   x2    = X { a: 1, b: 2, ..DX };  // ok
    const X3: X = X { a: 1, b: 2, ..DX };  // error[E0016]
}
```

* [ ] It's weird that `X1` and `x2` are allowed, but not `X3`.
* [ ] Even if that's intentional, the error message seems off.

**Comment by jakubadamw** [comment]
The provided code compiles fine both in 1.36 and the 2019-08-12 nightly.

I did a quick search through the test suite and there's none for this particular case. Marking as such.

@rustbot modify labels: E-needstest
